pkg/fuzz/util.go

```diff
@@ -137,16 +137,31 @@ func (h Helper) T() test.Failer {
 	return h.t
 }
 
+type mutateCtx struct {
+	t        test.Failer
+	curDepth int
+	maxDepth int
+}
+
 // MutateStruct modify the field value of the structure.
 // It is mainly used to check the correctness of the deep copy.
-func MutateStruct(h Helper, st any) {
+func MutateStruct(t test.Failer, st any) {
+	ctx := mutateCtx{t: t, curDepth: 0, maxDepth: 100}
 	e := reflect.ValueOf(st).Elem()
-	if err := mutateStruct(h, e); err != nil {
-		h.t.Skip(err.Error())
+	if err := mutateStruct(ctx, e); err != nil {
+		// do not use t.Skip here, it will call runtime.Goexit()
+		t.Log(err.Error())
 	}
 }
 
-func mutateStruct(h Helper, e reflect.Value) error {
+func mutateStruct(ctx mutateCtx, e reflect.Value) error {
+	// this can happen when circular data structure
+	if ctx.curDepth >= ctx.maxDepth {
+		return fmt.Errorf("reach max depth and the mutation may not completed")
```

Comment:
If we return an error here, it will just be logged and ignored. Another option is to panic here